### PR TITLE
mtls checker handles dr explicitly setting ISTIO_MUTUAL

### PR DESCRIPTION
**Pull request by stevenctl**
cc @howardjohn

### CODE REVIEW

File: pilot/pkg/xds/ep_filters_test.go [hunk 1]
```diff
@@ -423,11 +452,20 @@ func TestEndpointsByNetworkFilter_WithConfig(t *testing.T) {
 	for configType, cases := range cases {
 		t.Run(configType, func(t *testing.T) {
 			for name, pa := range cases {
+				if name != "mtls-on-override-pa" {
```
Comment (howardjohn) [inline pilot/pkg/xds/ep_filters_test.go:455]: remove this?

**Review by howardjohn — APPROVED** [review]

**Comment by howardjohn** [comment]
feel free to remove hold once you fix the debug stuff. Thanks!

**Review by adiprerepa — APPROVED** [review]

**Pull request merged**